src/Elasticsearch.Net/Transport/Pipeline/RequestData.cs

```diff
@@ -151,6 +160,37 @@ IMemoryStreamFactory memoryStreamFactory
 
 		public override string ToString() => $"{Method.GetStringValue()} {_path}";
 
+		public static string DefaultJsonBasedOnConfigurationSettings(IConnectionConfigurationValues settings) =>
+			settings.EnableApiVersioningHeader ? MimeType : MimeTypeOld;
+
+
+		public static bool IsJsonMimeType(string mimeType) =>
+			ValidResponseContentType(MimeType, mimeType) || ValidResponseContentType(MimeTypeOld, mimeType);
+
+		public static bool ValidResponseContentType(string acceptMimeType, string responseMimeType)
+		{
+			if (string.IsNullOrEmpty(acceptMimeType)) return false;
+			if (string.IsNullOrEmpty(responseMimeType)) return false;
+
+			// we a startswith check because the response can return charset information
+			// e.g: application/json; charset=UTF-8
+			if (acceptMimeType == RequestData.MimeTypeOld)
+				return responseMimeType.StartsWith(RequestData.MimeTypeOld);
```

Comment:
May want to force this as Ordinal-based since StartsWith and EndsWith are annoying culture-aware by default.